### PR TITLE
KubeVirt: VMIs with Service

### PR DESCRIPTION
**Pull request by fabiand**
## Description

Introduce a new example for KubeVirt VMs in order to create a Service in order to expose an SSH service, whenever a VMI is getting created.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

**Comment by fabiand** [comment]
@chipzoller mind reviewing this one as well?

(It passes locally)

File: kubevirt/add-services/add-services.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      namespace: "{{request.object.metadata.namespace}}"
+      synchronize: true
+      data:
+        metadata:
+          ownerReferences:
+          - apiVersion: kubevirt.io/v1
+            kind: VirtualMachineInstance
+            name: "{{request.object.metadata.name}}"
+            uid: "{{request.object.metadata.uid}}"
+        spec:
```
Comment (chipzoller) [inline kubevirt/add-services/add-services.yaml:36]: Just a suggestion, but I might force the type `ClusterIP` here in my definition so it's crystal clear what will be created without relying upon field/value defaults of the API server.

**Review by chipzoller — CHANGES_REQUESTED** [review]
Thanks for another KubeVirt sample! Looks good with a couple minor comments. Any other KubeVirt policy ideas you have, they'd be great to have.

File: kubevirt/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+This directory contains a few examples in order to apply policies on
+[KubeVirt](https://kubevirt.io) VMs.
+
+Examples
+
+- add-services - Add a service for every VMI which is getting created.
+  This is an example for a validating policy.
+- enforce-instancetype - Enforce the use of instance types, prevent the creation of "custom" VMs
+  This is an example for a generating policy.
```
[the remark this answers is not in the document]
Comment (fabiand) [reply kubevirt/README.md:11]: Meh. Absolutely, thanks! Fixed.

File: kubevirt/add-services/add-services.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: k6t-add-services
+  annotations:
```
[the remark this answers is not in the document]
Comment (fabiand) [reply kubevirt/add-services/add-services.yaml:5]: Fixed

File: kubevirt/add-services/add-services.yaml (hunk 1 above)
Comment (fabiand) [reply kubevirt/add-services/add-services.yaml:36]: Agreed. Thanks for weighing in here.

**Comment by fabiand** [comment]
Thanks for the review.

> Any other KubeVirt policy ideas you have, they'd be great to have.

There are some more user use-cases which I think we can solve with Kyverno.

**Review by chipzoller — APPROVED** [review]

**Comment by chipzoller** [comment]
Thanks again!

**Pull request merged**